src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -359,33 +370,43 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 				}
 
 				const message = localize('workspaceRecommended', "This workspace has extension recommendations.");
-				const action = this.instantiationService.createInstance(ShowWorkspaceRecommendedExtensionsAction, ShowWorkspaceRecommendedExtensionsAction.ID, localize('showRecommendations', "Show Recommendations"));
+				const showAction = this.instantiationService.createInstance(ShowWorkspaceRecommendedExtensionsAction, ShowWorkspaceRecommendedExtensionsAction.ID, localize('showRecommendations', "Show Recommendations"));
+				const installAllAction = this.instantiationService.createInstance(InstallWorkspaceRecommendedExtensionsAction, InstallWorkspaceRecommendedExtensionsAction.ID, localize('installAll', "Install All"));
```

Comment:
I would assume this action is shown only when there are extensions to install?

Reply:
The pop up itself is shown only if there are workspace recommendations to install. There is a check 4 lines above which checks this.